File: DSRFQ.Web/Modules/ServerTypes/Costing/CostingPartBomResultsForm.ts
import { PrefixedContext, IntegerEditor, StringEditor, DecimalEditor, DateEditor, initFormType } from '@serenity-is/corelib';

export interface CostingPartBomResultsForm {
    CostingPartId: IntegerEditor;
    PartNumber: StringEditor;
    Description: StringEditor;
    Quantity: DecimalEditor;
    InternalEngineeringNumber: StringEditor;
    IsManual: IntegerEditor;
    InsertDate: DateEditor;
    InsertUserId: IntegerEditor;
    UpdateDate: DateEditor;
    UpdateUserId: IntegerEditor;
    DeleteDate: DateEditor;
    DeleteUserId: IntegerEditor;
    IsActive: IntegerEditor;
}

export class CostingPartBomResultsForm extends PrefixedContext {
    static readonly formKey = 'Costing.CostingPartBomResults';
    private static init: boolean;
    
    constructor(prefix: string) {
        super(prefix);
        if (!CostingPartBomResultsForm.init)  {
            CostingPartBomResultsForm.init = true;
            
            var w0 = IntegerEditor;
            var w1 = StringEditor;
            var w2 = DecimalEditor;
            var w3 = DateEditor;

            initFormType(CostingPartBomResultsForm, [
            'CostingPartId', w0,
            'PartNumber', w1,
            'Description', w1,
            'Quantity', w2,
            'InternalEngineeringNumber', w1,
            'IsManual', w0,
            'InsertDate', w3,
            'InsertUserId', w0,
            'UpdateDate', w3,
            'UpdateUserId', w0,
            'DeleteDate', w3,
            'DeleteUserId', w0,
            'IsActive', w0,
            ]);
        }
    }
}

File: DSRFQ.Web/Modules/Costing/CostingPartBomResults/CostingPartBomResultsForm.cs
using Serenity.ComponentModel;
using System;

namespace DSRFQ.Costing.Forms;

[FormScript("Costing.CostingPartBomResults")]
[BasedOnRow(typeof(CostingPartBomResultsRow), CheckNames = true)]
public class CostingPartBomResultsForm
{
    public int CostingPartId { get; set; }
    public string PartNumber { get; set; }
    public string Description { get; set; }
    public decimal Quantity { get; set; }
    public string InternalEngineeringNumber { get; set; }
    public bool IsManual { get; set; }
}